samza-kafka/src/main/scala/org/apache/samza/config/KafkaConfig.scala

```diff
@@ -281,19 +336,22 @@ class KafkaConfig(config: Config) extends ScalaMapConfig(config) {
   // topic.
   def getCheckpointTopicProperties() = {
     val segmentBytes: Int = getCheckpointSegmentBytes()
+    val maxMessageBytes: Int = getCheckpointMaxMessageBytes()
     val appConfig = new ApplicationConfig(config)
     val isStreamMode = appConfig.getAppMode == ApplicationMode.STREAM
     val properties = new Properties()
 
     if (isStreamMode) {
       properties.putAll(ImmutableMap.of(
         "cleanup.policy", "compact",
-        "segment.bytes", String.valueOf(segmentBytes)))
+        "segment.bytes", String.valueOf(segmentBytes),
+        "max.message.bytes", String.valueOf(maxMessageBytes)))
     } else {
       properties.putAll(ImmutableMap.of(
         "cleanup.policy", "compact,delete",
         "retention.ms", String.valueOf(KafkaConfig.DEFAULT_RETENTION_MS_FOR_BATCH),
-        "segment.bytes", String.valueOf(segmentBytes)))
+        "segment.bytes", String.valueOf(segmentBytes),
+        "max.message.bytes", String.valueOf(maxMessageBytes)))
```

Comment:
Currently the checkpoint topic properties like `task.checkpoint.replication.factor`, `task.checkpoint.segment.bytes` are documented in samza [configuration-table](https://samza.apache.org/learn/documentation/latest/jobs/configuration-table.html). Also, if you agree it would be useful, then we can max.msg.bytes config with default to public configuration table.